### PR TITLE
add playground for debugging using vite and next.js

### PR DESCRIPTION
**Pull request by promer94**
https://user-images.githubusercontent.com/15878786/102889914-72fe8c80-4496-11eb-9636-f36067f36b0e.mov

```bash
yarn install
yarn register
yarn prepare:vite
yarn dev:vite
```

- For SSR

```bash
yarn install
yarn register
yarn prepare:next
yarn dev:next
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,4 @@ esm
 .vscode
 examples/**/yarn.lock
 package-lock.json
+playground/**
```
Comment (shuding) [inline .gitignore:13]: Oh I think it's not included in the PR?

Comment (promer94) [reply .gitignore:13]: ```yarn prepare:vite``` and ```yarn prepare:next``` will use git to get playground templates from [vite-swr](https://github.com/promer94/vite-swr.git) and [next-swr](https://github.com/promer94/next-swr).

So developers will not accidentally commit their changes to the playground

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+## Request for a New Feature
+
+For new features, it would be great to have some discussions from the community before starting working on it. You can either create an issue (if there isn't one) or post a thread in the [Discussions](https://github.com/vercel/swr/discussions) page to describe the feature that you want to have.
+
+If possible, you can add other additional context like how this feature can be implemented technically, what other alternative solutions we can have, etc.
+
+## Open a PR for Bugfix or Feature
+
+### Local Development
+
```
Comment (shuding) [inline CONTRIBUTING.md:24]:
```suggestion

To develop SWR locally, you can use the Vite SWR playground to play with the source code inside the browser. You can follow these steps:

```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+```bash
+yarn install
+yarn register
+yarn prepare:vite
+yarn dev:vite
+```
+
+- For SSR
```
Comment (shuding) [inline CONTRIBUTING.md:32]:
```suggestion
To test SSR related features, you need to use the Next.js SWR playground instead:
```

File: package.json [hunk 4]
```diff
@@ -15,15 +15,21 @@
   "homepage": "https://swr.vercel.app",
   "license": "MIT",
   "scripts": {
+    "dev:vite": "concurrently \"yarn watch:esm\" \"cd playground/vite-swr && yarn dev\"",
+    "dev:next": "concurrently \"yarn watch:esm\" \"cd playground/next-swr && yarn dev\"",
```
Comment (shuding) [inline package.json:19]:
```suggestion
```

File: package.json [hunk 5]
```diff
@@ -15,15 +15,21 @@
...
+    "watch:esm": "tsc --module ES6 --outDir esm --target esnext --watch",
     "types:check": "tsc --noEmit",
     "format": "prettier --write \"{src,test,examples}/**/*.{ts,tsx}\"",
     "lint": "eslint \"{src,test,examples}/**/*.{ts,tsx}\"",
     "lint:fix": "eslint \"{src,test,examples}/**/*.{ts,tsx}\" --fix",
-    "test": "jest"
+    "test": "jest",
+    "register": "yarn link && cd node_modules/react && yarn link",
+    "prepare:vite": "cd playground && git clone https://github.com/promer94/vite-swr && cd vite-swr && yarn && yarn link react && yarn link swr",
+    "prepare:next": "yarn build && cd playground && git clone https://github.com/promer94/next-swr && cd next-swr && yarn && yarn link react && yarn link swr"
```
Comment (shuding) [inline package.json:32]:
```suggestion
    "prepare:next": "yarn build && cd playground && git clone https://github.com/promer94/next-swr && cd next-swr && yarn && yarn link react && yarn link swr"
    "dev:vite": "concurrently \"yarn watch:esm\" \"cd playground/vite-swr && yarn dev\"",
    "dev:next": "concurrently \"yarn watch:esm\" \"cd playground/next-swr && yarn dev\"",
```

**Review by shuding — APPROVED** [review]
Thank you for the great work! 👍

**Pull request merged**